### PR TITLE
Add GeLU kernel that uses tanh approximation

### PR DESCRIPTION
**Pull request by sosborne-TT**
### Summary

Adds `ttnn.gelu(variant=Tanh)` that closely matches the equivalent torch function and updates some DiT models to use the new version.

The new kernel calculates `0.5·x·(1 + tanh(√(2/π)·(x + 0.044715·x³)))` in FP32, reusing `_sfpu_tanh_fp32_accurate_`. The legacy `fast_and_approximate_mode=True/False` bool overload still works and resolves to FastLut / Accurate.

A new test `tests/ttnn/unit_tests/operations/eltwise/test_gelu_fw_variants.py` checks per-element ULP against torch.nn.functional.gelu.

### Notes for reviewers

- In BF16 precision the new function matches the torch version except for 2 values with difference=1 ULP and except for denormalized values. In FP32 precision the new function matches within 400 ULP.
- There should be no change to existing code except for some DiT models: Wan 2.2, SD 3.5 and T5 (switched from accurate -> tanh GeLU) and CLIP and LTX which use the same GeLU variant but a different code path.

### CI

- [x] Wan T3K demo test [passes](https://github.com/tenstorrent/tt-metal/actions/runs/27800949622)
- [x] Wan T3K perf test [passes](https://github.com/tenstorrent/tt-metal/actions/runs/27848667169)
- [ ] Wan Galaxy demo test -- failing on main
- [ ] Wan Galaxy perf test

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:sosborne/gelu-kernel-update)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:sosborne/gelu-kernel-update)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:sosborne/gelu-kernel-update)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:sosborne/gelu-kernel-update)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:sosborne/gelu-kernel-update)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:sosborne/gelu-kernel-update)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:sosborne/gelu-kernel-update)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:sosborne/gelu-kernel-update)

### CODE REVIEW

File: models/tt_dit/layers/linear.py [hunk 1]
```diff
@@ -17,6 +15,19 @@
     ttnn.float32: ttnn.MathFidelity.HiFi4,
 }
 
+# Activation strings accepted by Linear / ColParallelLinear `activation_fn`,
+# mapped to the (UnaryOpType, param0_float) tuples the matmul fused-activation path expects.
+#
+# "gelu":      exact GELU (piecewise CDF / FP32 erf), matches F.gelu().
+# "gelu_fast": 6-segment piecewise-linear LUT, ~1% absolute error vs exact GELU.
+# "gelu_tanh": FP32 tanh approximation, matches F.gelu(approximate="tanh"). NaN-safe
+#              (kernel uses x*x*x for the cube, no exp/log path).
```
[the remark this answers is not in the document]
Comment (sosborne-TT) [reply models/tt_dit/layers/linear.py]: Addressed

**Review by bbradelTT — APPROVED** [review]

**Review by ndivnicTT — APPROVED** [review]

**Comment by ndivnicTT** [comment]
@sosborne-TT FYI, we have both accuracy and perf SFPU testing in tt-llk and you can consider plumbing new kernel into it. @jmacanovicTT is wrapping up this PR: https://github.com/tenstorrent/tt-metal/pull/47249 which is very useful for testing SFPU kernels.

File: tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_gelu.h [hunk 2]
```diff
@@ -347,6 +348,68 @@ inline void calculate_gelu() {
...
+
+        // x + (0.044715 * x^3).
+        sfpi::vFloat kx3 = x3 * GELU_TANH_K;
+        sfpi::vFloat inner = x + kx3;
+
+        sfpi::vFloat scaled = inner * SQRT_2_OVER_PI;
+
+        // Handle +-0 using the sign of x to prevent 1 ULP difference.
+        sfpi::vFloat zero = sfpi::vConst0;
+        sfpi::vFloat result = sfpi::copysgn(zero, x);
```
Comment (amahmudTT) [inline tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_gelu.h:383]: Cant vConst0 be used directly, afaik, the compiler here is not smart enough to ignore it.

**Review by s-jovic — APPROVED** [review]

File: tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_gelu.h (hunk 2 above)
Comment (sosborne-TT) [reply tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_gelu.h:383]: Made the change.

**Review by cglagovichTT — APPROVED** [review]

**Review by Aswinmcw — APPROVED** [review]
LGTM

**Review by philei-tt — APPROVED** [review]
(nit) Usually It's also good to test that op cache is properly reused. But since this is basically an extension of unary op, I guess it doesn't matter. 

I think matching torch api would be more intuitive and look nicer (ttnn.gelu(inputs, approximate="tanh"), instead of ttnn.gelu_tanh(inputs)), but I'm not sure if this is possible with current unary op infra

**Comment by sosborne-TT** [comment]
Thanks @philei-tt. One thing that will work with these changes is: `y = ttnn.gelu(x, variant=ttnn.GeluVariant.Tanh)`.
